elide-datastore/elide-datastore-aggregation/src/test/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/expression/JoinExpressionExtractorTest.java

```diff
@@ -0,0 +1,541 @@
+
+    // Both dim4 and dim5 passes same value for join expression's argument 'exprArg'
+    // Single Join expression is generated.
+    @Test
+    void testArgumentsInJoinExprCase1() {
+
+        Map<String, Argument> dim4Arg = new HashMap<>();
+        dim4Arg.put("exprArg", Argument.builder().name("exprArg").value("same").build());
+        SQLDimensionProjection dim4 = (SQLDimensionProjection) table.getDimensionProjection("dim4", "dim4", dim4Arg);
+
+        Map<String, Argument> dim5Arg = new HashMap<>();
+        dim5Arg.put("exprArg", Argument.builder().name("exprArg").value("same").build());
+        SQLDimensionProjection dim5 = (SQLDimensionProjection) table.getDimensionProjection("dim5", "dim5", dim5Arg);
+
+        Query query = Query.builder()
+                        .source(table)
+                        .dimensionProjection(dim4)
+                        .dimensionProjection(dim5)
+                        .arguments(emptyMap())
+                        .build();
+
+        String generatedSql = engine.explain(query).get(0);
+
+        String expectedSQL =
+                          "SELECT " + NL
+                        + "  DISTINCT `MainTable_join2_XXX`.`dim1` AS `dim4`," + NL
+                        + "           `MainTable_join2_XXX`.`dim1` AS `dim5` " + NL
+                        + "FROM " + NL
+                        + "  `main_table` AS `MainTable` " + NL
+                        + "LEFT OUTER JOIN " + NL
+                        + "  `join_table` AS `MainTable_join2_XXX` " + NL
+                        + "ON " + NL
+                        + "  value = 'same' AND `MainTable`.`dim6` - 'bar' = `MainTable_join2_XXX`.`dim4` - 'foo' ";
+
+        assertEquals(formatExpected(expectedSQL), formatGenerated(generatedSql));
+    }
+
+    // dim4 and dim5 passes different value for join expression's argument 'exprArg'
+    // 2 Join expressions are generated.
+    @Test
+    void testArgumentsInJoinExprCase2a() {
+
+        Map<String, Argument> dim4Arg = new HashMap<>();
+        dim4Arg.put("exprArg", Argument.builder().name("exprArg").value("value4").build());
+        SQLDimensionProjection dim4 = (SQLDimensionProjection) table.getDimensionProjection("dim4", "dim4", dim4Arg);
+
+        Map<String, Argument> dim5Arg = new HashMap<>();
+        dim5Arg.put("exprArg", Argument.builder().name("exprArg").value("value5").build());
+        SQLDimensionProjection dim5 = (SQLDimensionProjection) table.getDimensionProjection("dim5", "dim5", dim5Arg);
+
+        Query query = Query.builder()
+                        .source(table)
+                        .dimensionProjection(dim4)
+                        .dimensionProjection(dim5)
+                        .arguments(emptyMap())
+                        .build();
+
+        String generatedSql = engine.explain(query).get(0);
+
+        String expectedSQL =
+                          "SELECT " + NL
+                        + "  DISTINCT `MainTable_join2_XXX`.`dim1` AS `dim4`," + NL
```

Comment:
Is there any way we can tell the XXX aliases are different here?

Reply:
using the original values.